azure/services/agentpools/agentpools.go

```diff
@@ -122,6 +123,7 @@ func (s *Service) Reconcile(ctx context.Context) error {
 				EnableAutoScaling:   profile.EnableAutoScaling,
 				MinCount:            profile.MinCount,
 				MaxCount:            profile.MaxCount,
+				NodeLabels:          profile.NodeLabels,
```

Comment:
are there any existing unit tests that can be modified to test the behavior this PR fixes?

Reply:
I think the agentpool reconcile test could be modified for this. I will update the PR

Reply:
added a separate test for normalized diff, as I expect that test to grow when we keep adding more changes with few immutable and some mutable properties to the agentpool